decidim-assemblies/app/models/decidim/assembly.rb

```diff
@@ -27,12 +27,21 @@ class Assembly < ApplicationRecord
 
     has_many :features, as: :participatory_space, dependent: :destroy
 
+    has_many :assembly_private_users, class_name: "Decidim::AssemblyPrivateUser", foreign_key: "decidim_assembly_id", dependent: :destroy
+    has_many :users, through: :assembly_private_users, class_name: "Decidim::User", foreign_key: "decidim_user_id"
+
     validates :slug, uniqueness: { scope: :organization }
     validates :slug, presence: true, format: { with: Decidim::Assembly.slug_format }
 
     mount_uploader :hero_image, Decidim::HeroImageUploader
     mount_uploader :banner_image, Decidim::BannerImageUploader
 
+    scope :private_spaces_user, lambda { |user|
+      joins("LEFT JOIN decidim_assembly_private_users ON
+             decidim_assembly_private_users.decidim_assembly_id = decidim_assemblies.id")
+        .where("(private_space = true and decidim_assembly_private_users.decidim_user_id
+             = #{user} ) or private_space = false")
```

Comment:
Don't interpolate strings here, use `?` to replace the values: `.where("my_field = ?", my_value)`